### ISSUE TITLE
load_model() does not return when layer of model was renamed

### ISSUE THREAD

**Issue by blubbrezn** [keras-team/keras#14356]
keras version 2.2.4 , I also tried 2.4.3
tensorflow 2.2.0
Windows 10 Pro Build 19041

I cannot open a saved model, load_model() newer returns, it runs for ever without displaying an error.
It hapens when the Normalisation layer (not BatchNormalisation) is renamed. Renaming other layers like the dense, doesnt trigger this error.
I tried to integrate the information that the Normalisation layer is initalized into the model that no aditional information is needed. 
Is there a posibitly to see if a Normalisation layer is allready initialized (adapt())? 
Is there a posibility to load these models somehow or are they lost?

The following code demonstrates my problem:
The first load works, but the second just leads to high CPU load.
I get this output form the Code:
```
INFO:tensorflow:Assets written to: test.nn\assets
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
load sucessfully model
INFO:tensorflow:Assets written to: test.nn\assets
```
Code:
```
from tensorflow.keras.layers import Dense, Input
from tensorflow.keras.layers.experimental.preprocessing import Normalization
from tensorflow.keras.models import Model, load_model
input_tensor = Input(12)
o = Normalization(name="uninitialized_input_normalisation")(input_tensor)
o = Dense(units=10, activation='relu', use_bias=True)(o)
o = Dense(2, use_bias=True)(o)
m = Model(inputs=input_tensor, outputs=o)
m.save("test.nn")
m1 = load_model("test.nn")
print(f"load sucessfully {m1.name}")
for layer in m.layers:
    if hasattr(layer, 'adapt') and layer.name == "uninitialized_input_normalisation":
        layer._name = "input_normalisation"
m.save("test.nn")
m1 = load_model("test.nn")
print(f"load sucessfully {m1.name}")
```
Thanks in advance

**Comment by jvishnuvardhan** [comment]
@blubbrezn I think once you rename any layer, you need to reconnect layers as shown below. Otherwise, Functional model will be looking for old-name ("uninitialized_input_normalisation") which doesn't exist anymore. Just added one line to your code.

```
for layer in m.layers:
    if hasattr(layer, 'adapt') and layer.name == "uninitialized_input_normalisation":
        layer._name = "input_normalisation"

# connecting renamed layers
m = Model(inputs=m.input, outputs=m.output)
```

[Here](https://colab.research.google.com/gist/jvishnuvardhan/7660719a501b9eaa8ca11292a5d8d6c9/untitled.ipynb) is a gist for a reference. Thanks!
